README.md

```diff
@@ -14,8 +10,279 @@ But we need have a clear way to discover how you've been invited to a group with
+
+```
+/v1/:shard/:groupKey
+```
+Where:
+- `:shard` is the shard feed derived using BFE encoded `groupKey`
+    <details>
+      <summary>details</summary>
+      <div>
+        We cannot use the `groupId`, as this is publicly known, which would give attackers a way to test if people are in the group (breaking Principle 1.)
+        <br />
+        We choose the the `groupKey` because it is a value known only to those already in the group.
+      </div>
+    </details>
+- `:groupKey` is a content feed where
+    - `feedpurpose = groupKey` where `groupKey` is the ssb-uri encoded groupKey for the group
+    - `feedtype = ?` :warning:
+    - the announcement of the this sub-feed MUST be encrypted with this group's `groupKey`
+      <details>
+        <summary>details</summary>
+        <div>
+          We need a `feedpurpose` which is unique to the group, which the `groupKey` is.
+          We cannot use the `groupId`, because this is derived using the group init message, which does not exist until our feed exists.
+          We encrypt this announce message so as not to leak the groupKey AND to protect group membership.
+          <br />
+          For sympathetic replication we will therefor need a distinct type of announce message (TODO)
+        </div>
+      </details>
+
+
+## Flows
+
+### 1. Creating a group
+
+Staltz starts up his application.
+We assume he has already created his `group-invite` feed (following the spec above).
+In his application he creates a new "helsinki" group, which means he:
+1. creates a new symmetric `groupKey`
+2. create a content feed under a new shard (using the `groupKey` following the spec above)
+3. publishes an encrypted `group/init` message to that new "helsinki" feed
+4. publishes an encrypted `group/add-member` message to his "group-invite" feed
+      <details>
+        <summary>details</summary>
+        <div>
+          This helps new members quickly see he is a member of the group, and also ensures he has a copy of the groupKey persisted in his records (encrypted to him and the group)
+        </div>
+      </details>
+
+```mermaid
+graph TB
+
+root(root)
+v1(v1)
+4(4)
+d(d)
+group-invite(group-invite):::groupInvite
+helsinki(helsinki):::group
+
+subgraph Staltz
+  root --> v1 --> 4 --> helsinki
+           v1 --> d --> group-invite
+end
+
+classDef default stroke:none;
+classDef groupInvite fill: #BF2669, stroke:none, color:white;
+classDef group fill: #702A8C, stroke: none, color:white;
+```
+_Diagram showing Staltz feed state from his perspective_
+
+### 2. Group creator invites someone
+
+Staltz wants to invite his friend Arj to the group he set up, so he publishes a `group/add-member` message (which conatins the `groupKey`) to his "group-invite" feed.
+
+When Arj next stats up his application and replicates Staltz's feed (they are friends) he discovers the new `group/add-member` for him Staltz's "group-invite" feed (peers MUST replicate friends group-invite feeds).
```

Comment:
```suggestion
When Arj next starts up his application and replicates Staltz's feed tree (they are friends), he discovers the new `group/add-member` for him on Staltz's "invitations" feed (because peers must replicate their friends' "invitations" feeds).
```